pkg/pmem-csi-driver/registryserver.go

```diff
@@ -66,6 +70,17 @@ func (rs *registryServer) RegisterController(ctx context.Context, req *registry.
 	if req.GetEndpoint() == "" {
 		return nil, fmt.Errorf("Missing endpoint address")
 	}
+
+	if rs.clientTLSConfig != nil {
+		peer, err := getPeer(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if peer != "pmem-node-controller" {
+			return nil, status.Errorf(codes.PermissionDenied, "caller '%s' is not a valid node controller", peer)
+		}
+	}
```

Comment:
It is possible to set `ServerName` also in the `tls.Config` for a gRPC server and then gRPC itself does this check. Better use that, it should be simpler.

Reply:
The only works if all clients have the same common name, but that is the case here.

Reply:
> It is possible to set `ServerName` also in the `tls.Config` for a gRPC server and then gRPC itself does this check. Better use that, it should be simpler.

I tried setting registry server's  tls.Config.ServerName  to "pmem-registry" but couldn't find the wrong(non pmem-node-controller) peers. But i am not sure how could it user server.ServerName to detect wrong peer by. Am i missing something?

Reply:
> That only works if all clients have the same common name, but that is the case here.

Yes, all Nodes should connect to RegistryServer using same common name : pmem-node-controller.

Reply:
Oops, you are right. I thought the `ServerName` was used during client certificate validation already, but that's not the case. One has to write a custom `VerifyPeerCertificate` function. I can show you in some Intel-internal code how that works. I still think it is better to do the check at the connection layer instead of adding it to every application method.

Reply:
@pohly Yep, adding VerifyPeerCertificate() function for server is much better solution, thanks for the pointer.

Reply:
now i added VerifyPeerCertificate for server tls config to validate right peer clients. Can you please have a look if any more changes needed.